sda_download/sda_download_test.go

```diff
@@ -0,0 +1,176 @@
+}
+
+func (suite *TestSuite) TestNoFiles() {
+
+	confPath := createConfigFile("s3cmd-download.conf", suite.accessToken)
+
+	os.Args = []string{"sda-download", "-dataset", "TES01", "-config", confPath.Name(), "-url", "https://some/url"}
+
+	err := SdaDownload(os.Args)
+	assert.EqualError(suite.T(), err, "no files to download")
+}
+
+func (suite *TestSuite) TestInvalidUrl() {
+
+	confPath := createConfigFile("s3cmd.conf", suite.accessToken)
+
+	os.Args = []string{"sda-download", "-dataset", "TES01", "-config", confPath.Name(), "-url", "https://some/url", "file1", "file2"}
+
+	err := SdaDownload(os.Args)
+	assert.Contains(suite.T(), err.Error(), "failed to get files, reason: failed to get response, reason: Get \"https://some/url/metadata/datasets/TES01/files\": dial tcp: lookup some")
+}
+
+func (suite *TestSuite) TestGetBody() {
+	// Create a test server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		// Set the response status code
+		w.WriteHeader(http.StatusOK)
+		// Set the response body
+		fmt.Fprint(w, "test response")
+	}))
+	defer server.Close()
+
+	// Make a request to the test server
+	body, err := getBody(server.URL, "test-token")
+	if err != nil {
+		suite.T().Errorf("getBody returned an error: %v", err)
+	}
+
+	// Check the response body
+	expectedBody := "test response"
+	if string(body) != expectedBody {
+		suite.T().Errorf("getBody returned incorrect response body, got: %s, want: %s", string(body), expectedBody)
+	}
+}
+
+func (suite *TestSuite) TestDownloadUrl() {
+	// Mock getBody function
+	defer func() { getResponseBody = getBody }()
+	getResponseBody = func(_, _ string) ([]byte, error) {
+		return []byte(`[
+            {
+                "fileId": "file1id",
+				"datasetId": "TES01",
+				"displayName": "file1",
+                "filePath": "path/to/file1",
+				"fileName": "4293c9a7-re60-46ac-b79a-40ddc0ddd1c6"
+            }
+        ]`), nil
+	}
+
+	// Test with valid base_url, token, dataset, and filename
+	baseURL := "https://some/url"
+	token := suite.accessToken
+	dataset := "TES01"
+	filepath := "path/to/file1"
+	expectedURL := "https://some/url/files/file1id"
+	url, err := getFileIDURL(baseURL, token, dataset, filepath)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), expectedURL, url)
+
+	// Test with filename not in response
+	filepath = "path/to/file2"
+	_, err = getFileIDURL(baseURL, token, dataset, filepath)
+	assert.Error(suite.T(), err)
+
+	// Test with fileID
+	filepath = "file1id"
+	_, err = getFileIDURL(baseURL, token, dataset, filepath)
+	assert.NoError(suite.T(), err)
+
+	// Testr with bad URL
+	_, err = getFileIDURL("some/url", token, dataset, filepath)
+	assert.Error(suite.T(), err)
```

Comment:
Add a test where the dataset name is an URL

Reply:
Done